m3/ice_simulator.py

```diff
@@ -235,6 +248,73 @@ def send_snoop(addr, data, control):
 
         logger.info("Replay finished.")
 
+    #
+    #
+    #
+    def transaction_thread(self):
+        ''' 
+        Replays a series of ICE transactions with timing information
+        '''
+        def read_raw_message():
+            msg_type, event_id, length = self.s.read(3)
+            length_int = ord(length)
+            logger.debug("Got a message of type: " + msg_type + 
+                    ' length: ' + str(length_int))
+            msg = self.s.read(length_int)
+
+            return msg_type + event_id + length + msg
+
+        logger.info("Transaction beginning")
+        last_ts = None
+
+        for line in open( self.args.transaction.strip() ):
+            
+            line = line.strip('\n')
+            line = line.strip('\r')
+
+            logger.info('Working on: ' + line)
+
+            if len(line) == 0: continue
+            elif line[0] in ['#', ' ', '/', ]: continue
+
+            elif line.startswith('WAIT'):
+               
+                #find the number after T
+                hex_tex = line.replace(' ','').split('T')[1].strip()
```

Comment:
Not that it matters much, but at some point if you add a command with a duplicate letter, this won't work as written, you can also do this in Python:

    >>> 'WAIT 0x50'.split('WAIT')[1].strip()
    '0x50'